### PR TITLE
implemented new scanner

### PR DESCRIPTION
**Pull request by ironoa**
https://github.com/w3f/SecOps/issues/102

### CODE REVIEW

File: charts/polkadot-watcher-transaction/values.yaml [hunk 1]
```diff
@@ -2,26 +2,22 @@ environment: production
...
-      balanceChange:
-        enabled: true
-      transferExtrinsic:
-        enabled: false
+      transferEventScanner:
+      enabled: true
+      sent: true
+      received: true
+      dataDir: "/app/data"
+      scanInterval: 600000
```
Comment (Nexus2k) [inline charts/polkadot-watcher-transaction/values.yaml:20]: I'd recommend to rename scanInterval field to scanIntervalSecs or some unit that people directly understand what unit those 600000 mean.

Comment (Nexus2k) [reply charts/polkadot-watcher-transaction/values.yaml:20]: Here iiuc a good name would be scanIntervalMilliSecs

File: src/subscriptionModules/eventScannerBased.ts [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        if(!result){
+          // better to risk to send them all again (block) than loose one of them
+          this.logger.warn(`\n*****\nSCAN had an issue at block ${blockNumber}, exiting...\n*****`)
+          throw `SCAN ERROR at block ${blockNumber}`
+        }
+      }
+      
+      if(result){
+        this.logger.info(`\n*****\nSCAN completed at block ${await this._getLastCheckedBlock()}\n*****`)
+      }else{
```
Comment (Nexus2k) [inline src/subscriptionModules/eventScannerBased.ts:156]: I don't believe this code path is ever reachable, therefore not needed. Either you throw at line 150 or result is still true at this point.

**Review by Nexus2k — APPROVED** [review]

File: src/subscriptionModules/eventScannerBased.ts (hunk 2 above)
Comment (ironoa) [reply src/subscriptionModules/eventScannerBased.ts:156]: True, I wanted to be really sure =)

**Review by lamafab — APPROVED** [review]

**Pull request merged**